ios/browser/ads/ads_service_impl_ios.mm

```diff
@@ -0,0 +1,237 @@
+  CHECK(IsRunning());
+
+  ads_->TriggerInlineContentAdEvent(placement_id, creative_instance_id,
+                                    mojom_ad_event_type, std::move(callback));
+}
+
+void AdsServiceImplIOS::TriggerNewTabPageAdEvent(
+    const std::string& placement_id,
+    const std::string& creative_instance_id,
+    mojom::NewTabPageAdEventType mojom_ad_event_type,
+    TriggerAdEventCallback callback) {
+  CHECK(IsRunning());
+
+  ads_->TriggerNewTabPageAdEvent(placement_id, creative_instance_id,
+                                 mojom_ad_event_type, std::move(callback));
+}
+
+void AdsServiceImplIOS::MaybeGetNotificationAd(
+    const std::string& placement_id,
+    MaybeGetNotificationAdCallback callback) {
+  CHECK(IsRunning());
+
+  ads_->MaybeGetNotificationAd(placement_id, std::move(callback));
+}
+
+void AdsServiceImplIOS::TriggerNotificationAdEvent(
+    const std::string& placement_id,
+    mojom::NotificationAdEventType mojom_ad_event_type,
+    TriggerAdEventCallback callback) {
+  CHECK(IsRunning());
+
+  ads_->TriggerNotificationAdEvent(placement_id, mojom_ad_event_type,
+                                   std::move(callback));
+}
+
+void AdsServiceImplIOS::TriggerPromotedContentAdEvent(
+    const std::string& placement_id,
+    const std::string& creative_instance_id,
+    mojom::PromotedContentAdEventType mojom_ad_event_type,
+    TriggerAdEventCallback callback) {
+  CHECK(IsRunning());
+
+  ads_->TriggerPromotedContentAdEvent(placement_id, creative_instance_id,
+                                      mojom_ad_event_type, std::move(callback));
+}
+
+void AdsServiceImplIOS::MaybeGetSearchResultAd(
+    const std::string& placement_id,
+    MaybeGetSearchResultAdCallback callback) {
+  CHECK(IsRunning());
+
+  ads_->MaybeGetSearchResultAd(placement_id, std::move(callback));
+}
+
+void AdsServiceImplIOS::TriggerSearchResultAdEvent(
+    mojom::CreativeSearchResultAdInfoPtr mojom_creative_ad,
+    mojom::SearchResultAdEventType mojom_ad_event_type,
+    TriggerAdEventCallback callback) {
+  CHECK(IsRunning());
+
+  ads_->TriggerSearchResultAdEvent(std::move(mojom_creative_ad),
+                                   mojom_ad_event_type, std::move(callback));
+}
+
+void AdsServiceImplIOS::PurgeOrphanedAdEventsForType(
+    mojom::AdType mojom_ad_type,
+    PurgeOrphanedAdEventsForTypeCallback callback) {
+  CHECK(IsRunning());
+
+  ads_->PurgeOrphanedAdEventsForType(mojom_ad_type, std::move(callback));
+}
+
+///////////////////////////////////////////////////////////////////////////////
+
+void AdsServiceImplIOS::Shutdown() {
+  Cleanup();
+}
+
+void AdsServiceImplIOS::InitializeDatabase(const std::string& storage_path) {
+  storage_path_ = base::FilePath(storage_path);
+
+  database_ = base::SequenceBound<brave_ads::Database>(
+      database_queue_,
+      base::FilePath(storage_path_.Append(kAdsDatabaseFilename)));
+}
+
+void AdsServiceImplIOS::Cleanup() {
+  ads_.reset();
+  database_.Reset();
+}
```

Comment:
What are your thoughts on removing Cleanup and instead adding ads_.reset() and database_.reset() directly into Shutdown, then calling Shutdown directly in place of Cleanup?

Reply:
Agreed, done 👍